scripts/agent_bridge_provider.py

```
@@ -0,0 +1,92 @@
+            raise ProviderError("provider returned an invalid response") from error
+        if not isinstance(plan, dict):
+            raise ProviderError("provider returned an invalid response")
+        return plan
+
+
+def provider_from_environment(environment):
+    provider_name = environment.get("DAVOS_AGENT_PROVIDER", "fake").strip().lower()
+    if provider_name == "fake":
+        return FakeProvider()
+    if provider_name != "openai":
+        raise ProviderError("unsupported provider", "provider_config")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)